### ISSUE TITLE
MarkerViewOptions.icon() Should Be Optional

### ISSUE THREAD

**Issue by cammace** [mapbox/mapbox-gl-native#5276]
Position should be the only thing that needs to be passed to `MarkerViewOptions()`. Currently `.icon` is also required instead of just using the default red marker icon `MarkerOptions()` uses. In other words, setting up a simple marker view currently looks like this:

```java
 mapboxMap.addMarker(new MarkerViewOptions()
                        .position(new LatLng(-33.85699436, 151.21510684))
                        .icon(icon)); // This line shouldn't be required
```

cc: @bleege @tobrun @zugaldia @ivovandongen

**Comment by cammace** [comment]
BTW the [default marker image](https://github.com/mapbox/mapbox-gl-native/blob/master/platform/android/MapboxGLAndroidSDK/src/main/res/drawable-xxhdpi/default_marker.png) has a bunch of bottom padding right now and can either be cropped out or `.anchor(0.5f,0.5f)` needs to be used

**Comment by tobrun** [comment]
@cammace great catch on Icon. Couple of remarks on the subject. Yes we should provide a default icon when none is provided but note that we can't reuse the default one:

<img width="296" alt="screen shot 2016-06-08 at 10 53 16" src="https://cloud.githubusercontent.com/assets/2151639/15888499/6479e998-2d67-11e6-966a-95c5c560092e.png">


The default for gl markers contained extra space at the bottom because at that time we didn't have any setup for anchoring. We will need to crop that image and use that as the default for ViewMarkers. FWIW the default anchoring (conform to Google is at the bottom-middle = .anchor(0.5f,1.0f).

**Comment by Schumi09** [comment]
I really liked the explanations in #5079 by the way. When you are going to release 4.1.0 you should make users aware of this in the tutorials, especially because the anchoring differs for both types. I was pretty surprised at first some days ago ;)

**Comment by bleege** [comment]
Just noting that @Schumi09 has documented a good workaround for the current state of the SDK as well as inspiration for fixing this when we pick this issue back up in https://github.com/mapbox/mapbox-gl-native/issues/3276#issuecomment-224862363.  Thanks again @Schumi09!

**Comment by tobrun** [comment]
This landed in #5348 and was merged in `release-android-v4.1.0`. Closing
